tests/lax_scipy_special_functions_test.py

```diff
@@ -317,6 +317,31 @@ def testGammaIncCBoundaryValues(self):
     self._CheckAgainstNumpy(lsp_special.gammaincc, osp_special.gammaincc, args_maker, rtol=rtol)
     self._CompileAndCheck(lsp_special.gammaincc, args_maker, rtol=rtol)
 
+  def testBetaIncBoundaryValues(self):
+    dtype = jax.numpy.zeros(0).dtype  # default float dtype.
+    fi = jax.numpy.finfo(dtype)
+    nan = float('nan')
+    inf = float('inf')
+    tiny = fi.tiny
+    eps = fi.eps
+    # TODO(pearu): enable tiny samples when a fix to scipy/scipy#22682
+    # will be available
+    a_samples = [nan, -0.5, inf, 0, eps, 1, tiny][:-1]
+    b_samples = [nan, -0.5, inf, 0, eps, 1, tiny][:-1]
+    x_samples = [nan, -0.5, 0, 0.5, 1, 1.5]
+    if jtu.parse_version(scipy.__version__) >= (1, 16):
+      samples_slice = slice(None)
+    else:
+      # disable samples that contradict with scipy/scipy#22425
+      samples_slice = slice(None, len(a_samples) * len(b_samples) * 2)
```

Comment:
I find it very hard to understand the effect of this. From running the code, it looks like maybe we could get rid of `samples_slice` and instead say `a_samples = a_samples[:2]`: is that accurate?

Reply:
Specifying samples for different scipy versions is now simplified.